src/common/transformations/include/transformations/op_conversions/convert_ti_to_sequences.hpp

```diff
@@ -68,14 +70,29 @@ class ov::pass::ConvertTensorIteratorToSequence : public GraphRewrite {
     ConvertTensorIteratorToSequence();
 };
 
+class ov::pass::ConvertLoopWithSlicedInputConcatOutputToLSTMSequence : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("ConvertLoopWithSlicedInputConcatOutputToLSTMSequence", "0");
+    ConvertLoopWithSlicedInputConcatOutputToLSTMSequence();
+};
+
+class ov::pass::ConvertLoopWithScatterUpdateToLSTMSequence : public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("ConvertLoopWithScatterUpdateToLSTMSequence", "0");
+    ConvertLoopWithScatterUpdateToLSTMSequence();
+};
+
 /**
  * @ingroup ov_transformation_common_api
  * @brief Replaces Loop with LSTMCell inside to LSTMSequence
  */
-class ov::pass::ConvertLoopToLSTMSequence : public ov::pass::MatcherPass {
+class ov::pass::ConvertLoopToLSTMSequence : public ov::pass::GraphRewrite {
 public:
     OPENVINO_RTTI("ConvertLoopToLSTMSequence", "0");
-    ConvertLoopToLSTMSequence();
+    ConvertLoopToLSTMSequence() {
+        add_matcher<ov::pass::ConvertLoopWithScatterUpdateToLSTMSequence>();
+        add_matcher<ov::pass::ConvertLoopWithSlicedInputConcatOutputToLSTMSequence>();
```

Comment:
do we have any tests for this transformation?

Reply:
in progress

Reply:
Done